crates/sui-indexer-alt/src/lib.rs

```diff
@@ -441,101 +425,87 @@ pub async fn start_indexer(
         max_chunk_size: 5 * 300,
     });
 
-    indexer
-        .sequential_pipeline(
-            SumCoinBalances,
-            SequentialConfig {
-                committer: sum_coin_balances,
-                checkpoint_lag,
-            },
-        )
-        .await?;
-
-    indexer
-        .concurrent_pipeline(
-            WalCoinBalances,
-            ConcurrentConfig {
-                committer: wal_coin_balances,
-                pruner: pruner_config.clone(),
-            },
-        )
-        .await?;
-
-    indexer
-        .sequential_pipeline(
-            SumObjTypes,
-            SequentialConfig {
-                committer: sum_obj_types,
-                checkpoint_lag,
-            },
-        )
-        .await?;
-
-    indexer
-        .concurrent_pipeline(
-            WalObjTypes,
-            ConcurrentConfig {
-                committer: wal_obj_types,
-                pruner: pruner_config,
-            },
-        )
-        .await?;
-
-    // Other summary tables (without write-ahead log)
-    indexer
-        .sequential_pipeline(SumDisplays, sum_displays)
-        .await?;
-
-    indexer
-        .sequential_pipeline(SumPackages, sum_packages)
-        .await?;
-
-    // Unpruned concurrent pipelines
-    indexer.concurrent_pipeline(EvEmitMod, ev_emit_mod).await?;
-
-    indexer
-        .concurrent_pipeline(EvStructInst, ev_struct_inst)
-        .await?;
-
-    indexer
-        .concurrent_pipeline(KvCheckpoints, kv_checkpoints)
-        .await?;
-
-    indexer
-        .concurrent_pipeline(KvEpochEnds, kv_epoch_ends)
-        .await?;
-
-    indexer
-        .concurrent_pipeline(KvEpochStarts, kv_epoch_starts)
-        .await?;
+    let cancel = CancellationToken::new();
+    let retry_interval = ingestion.retry_interval();
+    let mut indexer = Indexer::new(db_args, indexer_args, ingestion, cancel.clone()).await?;
 
-    indexer.concurrent_pipeline(KvObjects, kv_objects).await?;
+    macro_rules! add_concurrent {
+        ($handler:expr, $config:expr) => {
+            indexer
+                .concurrent_pipeline($handler, $config.finish(&committer))
+                .await?
+        };
+    }
```

Comment:
I'm on the fence about readability for macros like this. On the one hand it really cuts down on noise and repetition but on the other I do think it can hurt readability of the code since you have to "run the macro yourself" when reading the code to truly understand whats happening. Not suggesting you should change this, this is more me thinking out loud...

Reply:
Yeah, I think in this commit in the stack, the macro is not super useful -- the impetus was that I was struggling to see which pipelines we had at a glance (and kind of pipeline they were), because the formatter had spread each one out ver three lines, but where it became more useful is later on when each macro invocation also handles enabling/disabling pipelines based on whether the config exists or not, and "finishing" the pipeline's arguments to apply overrides and check for unexpected fields -- at that point it becomes basically impossible to glean things at a glance.

But I do think it's worth documenting what these macros do so that folks don't need to read the implementation to run them in their head.